### ISSUE TITLE
fix: [feature_brick] Provider State Management gives a "Could not find the correct Provider<LoginNotifier> above this Consumer<LoginNotifier> Widget" error

### ISSUE THREAD

**Issue by ashishbeck-zeeve** [LukeMoody01/mason_bricks#19]
**feature_brick**

- [ ] model
- [x] feature_brick
- [ ] app_ui
- [ ] service
- [ ] service_package

**Description**
With the default implementation of provider state management, flutter throws this exception when `LoginPage()` is put in the widget tree.

**Steps To Reproduce**

1. Create a new project and a new feature brick
2. Put the feature page in home of `MaterialApp` of `main.dart`
```
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(),
      home: const LoginPage(),
   );
```
3. Build

**Expected Behavior**
It should display the correct page

**Solution**
Replace this in `login/view/login_page.dart`:
```
     return Provider(
      create: (context) =>
          ChangeNotifierProvider(create: (_) => LoginNotifier()),
      child: const Scaffold(
        body: LoginView(),
      ),
    );
```
with this:
```
    return ChangeNotifierProvider(
      create: (context) => LoginNotifier(),
      child: const Scaffold(
        body: LoginView(),
      ),
    );
```
or this:
```
    return MultiProvider(
      providers: [
        ChangeNotifierProvider(create: (_) => SettingsNotifier()),
      ],
      child: const Scaffold(
        body: LoginView(),
      ),
    );
```

**Additional Context**
Maybe I am implementing the widgets in the wrong way which is why I am getting this error. Please do let me know in case.

**Comment by LukeMoody01** [comment]
Hey @ashishbeck-zeeve 👋

Thanks for opening an issue! I will take a look at this as soon as possible!

**Comment by LukeMoody01** [comment]
Hey @ashishbeck-zeeve  👋

I looked into the issue and great pickup! Thanks for raising it.

I have published a fix and should now be available in version 0.5.1 of the feature_brick:
https://brickhub.dev/bricks/feature_brick/0.5.1
